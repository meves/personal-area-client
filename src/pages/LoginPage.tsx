import React from "react";
import styled from "styled-components";
import { LoginForm } from "../components/LoginForm";

const LoginPageWraper = styled.div`
    background-color: var(--login-page-bg-color);
`;

const LoginPage = () => {
    return (
        <LoginPageWraper className='centeredContainer flexDirectionColumn'>
            <LoginForm/>
        </LoginPageWraper>
    )
}

export default LoginPage;
