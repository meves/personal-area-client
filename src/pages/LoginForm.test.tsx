import React from "react";
import ReactDOM from "react-dom/client";
import LoginPage from "./LoginPage";

jest.mock("../components/LoginForm", () => {
    return function() {
        return (<form>LoginForm</form>)
    }
});

let root: any = null;
let container: any = null;
beforeEach(() => {
    container = document.createElement("div");
    root = ReactDOM.createRoot(container);
});
afterEach(() => {
    root.unmount(container);
    container.remove();
    container = null; 
});

describe("LoginForm test", () => {
    it("Login From renders", () => {
        root.render(<LoginPage/>);
        expect(container).toMatchSnapshot();
    });
});