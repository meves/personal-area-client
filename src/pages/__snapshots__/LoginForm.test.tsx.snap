// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm test Login From renders 1`] = `<div />`;
